src/Microsoft.Data/DataFrame.IO.cs

```diff
@@ -0,0 +1,261 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+
+namespace Microsoft.Data
+{
+
+    public partial class DataFrame
+    {
+        private static Type GuessKind(int col, List<string[]> read)
+        {
+            Type res = typeof(string);
+            int nbline = 0;
+            foreach (var line in read)
+            {
+                if (col >= line.Length)
+                    throw new FormatException(string.Format(Strings.LessColumnsThatExpected, nbline + 1)); // Will this really work?
```

Comment:
The question is w.r.t localization

Reply:
Yes, you can have placeholders in resources `{0}`.

---
In reply to: [299212735](https://github.com/dotnet/corefxlab/pull/2692#discussion_r299212735) [](ancestors = 299212735)